### ISSUE TITLE
Using -fPIE/-pie on macOS results in many spurious warnings

### ISSUE THREAD

**Issue by rgommers** [conda-forge/ctng-compiler-activation-feedstock#84]
When building a Python extension module on macOS (both x86-64 and arm64) with `clang` or `clang++`, you will see this warning for each extension ([example log from a conda-forge SciPy build](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=560378&view=logs&jobId=1b8be447-c2bd-5772-b66a-a1146441bf88&j=1b8be447-c2bd-5772-b66a-a1146441bf88&t=0da7bde9-21e8-5d19-f418-738876121ca7)):
```
ld: warning: -pie being ignored. It is only used when linking a main executable
```
When you build a package like SciPy, that's O(100) warnings. 

The problem isn't actually limited to macOS I think, I just don't see it on Linux because the `compilers` package pulls in GCC rather than Clang.

I have read https://conda-forge.org/docs/user/faq.html#faq-compiler-required-options, but I don't think adding `-fPIE` is one of the options that qualifies here - because it is not a flag that is valid for building Python extensions nor other types of binaries that aren't a main executable. Therefore I don't think it should be added by default.

I can imagine that the choice was made to only do this in Clang because macOS seemingly preferring position independent executables (https://en.wikipedia.org/wiki/Position-independent_code#Position-independent_executables) and warning on non-PIE binaries being uploaded to the App Store. That said, this should be dealt with in the build options for packages that are main executables though, and not for every compiler invocation.

**Comment by rgommers** [comment]
Apologies, wrong repo. Moving this to https://github.com/conda-forge/clang-compiler-activation-feedstock, closing.
